### PR TITLE
introduce constants section in MicroEJ Classpath

### PR DESCRIPTION
**Pull request by frivieremicroej**

### CODE REVIEW

File: ApplicationDeveloperGuide/classpath.rst [hunk 1]
```diff
@@ -218,6 +226,66 @@ For example, to define the property ``myProp`` with the value
...
+- String.
+
+Constants are declared in MicroEJ Classpath ``*.constants.list`` files. The file format is a
+standard Java properties file. Example:
+
+.. code-block:: xml
+   :caption: Example of Contents of a BON constants File
+
+   # The following property is embedded as a constant
+   com.mycompany.myconstantkey=com.mycompany.myconstantvalue
```
Comment (jerome-leroux) [inline ApplicationDeveloperGuide/classpath.rst:257]: This example is not displayed correctly on the Github preview.

File: ApplicationDeveloperGuide/classpath.rst [hunk 2]
```diff
@@ -218,6 +226,66 @@ For example, to define the property ``myProp`` with the value
...
+
+
+Constants are resolved at binary level without having to recompile the sources. 
+
+At link time, constants are directly inlined at the place of 
+``Constants.get[Type]`` method calls with no cost. 
+
+The String key parameter must be resolved as an inlined String:
+
+- either a String literal ``"com.mycompany.myconstantkey"``:
```
Comment (jerome-leroux) [inline ApplicationDeveloperGuide/classpath.rst:267]:
```suggestion
- either a String literal ``"com.mycompany.myconstantkey"``
```

File: ApplicationDeveloperGuide/classpath.rst [hunk 3]
```diff
@@ -218,6 +226,66 @@ For example, to define the property ``myProp`` with the value
...
+A boolean constant declared in an ``if`` statement condition can be used to fully remove portions of code.
+This feature is similar to C pre-processors ``#ifdef`` directive with the difference that this optimization is performed at binary level
+without having to recompile the sources.
+
+.. code-block:: java
+   :caption: Example of ``if`` code removal using a BON boolean constant
+
+   if (Constants.getBoolean("com.mycompany.myconstantkey")) {
+          System.out.println("this code and the constant string will be fully removed when the constant is resolved to 'false'")
+   }
```
Comment (jerome-leroux) [inline ApplicationDeveloperGuide/classpath.rst:281]: This example is not visible in Github preview

File: ApplicationDeveloperGuide/classpath.rst (hunk 1 above)
Comment (frivieremicroej) [reply ApplicationDeveloperGuide/classpath.rst:257]: Thanks. A note has been added to internal wiki for future RST coding rules. So far this is not blocker.

File: ApplicationDeveloperGuide/classpath.rst (hunk 3 above)
Comment (frivieremicroej) [reply ApplicationDeveloperGuide/classpath.rst:281]: Same comment as above.

**Pull request merged**